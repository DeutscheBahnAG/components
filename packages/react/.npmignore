# Documentation
*-doc.js
*-doc.scss
*-doc.css
# Tests
*.snap
*.test.*
#
.release-it.js
tsconfig.build.json
wss_config.yml
