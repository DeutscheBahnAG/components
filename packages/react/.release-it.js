module.exports = {
  git: {
    commitMessage: 'chore: Release @bahn-x/react@${version}',
    requireBranch: 'main',
    tagName: '@bahn-x/react@${version}',
  },
  npm: {
    publishPath: 'dist',
  },
};
