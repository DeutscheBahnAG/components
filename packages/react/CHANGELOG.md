# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Make sure to format this file using Prettier.

## [Unreleased]

## [1.1.1] - 2022-06-28

### Fixed

- Require latest design fixes of `@db-design/styles`

## [1.1.0] - 2022-06-28

### Added

- Columns support for Container

## [1.0.1] - 2022-05-11

### Fixed

- Missing inline spacer for `<Track>`

## [1.0.0] - 2022-05-10

### Changed

- Renamed `mt`/`mb` to `logoUnitsTop`/`logoUnitsBottom` for `<Logo>` and `<Pulse>`

## [1.0.0-pre.24] - 2022-03-23
