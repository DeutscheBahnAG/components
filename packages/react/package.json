{
  "name": "@db-design/react",
  "version": "1.0.0-pre.19",
  "description": "DB React Components",
  "repository": {
    "directory": "packages/react",
    "url": "git@ssh.git.tech.rz.db.de:bahn-x/webdev/web.git",
    "type": "git"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "dependencies": {
    "@db-design/react-icons": "^1.0.0-pre.1",
    "@db-design/styles": "^1.0.0-pre.17",
    "clsx": "^1.1.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-modal": "3.12.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.2",
    "@types/react-modal": "^3.10.6",
    "release-it": "^14.2.2",
    "typescript": "^4.1.3"
  },
  "scripts": {
    "build": "rm -rf dist && tsc --project tsconfig.build.json",
    "postbuild": "./postbuild.sh",
    "release": "release-it --disable-metrics"
  },
  "sideEffects": [
    "./dist/button/*",
    "./dist/modal/*",
    "./dist/pulse/*",
    "./dist/triptime/*",
    "./dist/triptimespan/*"
  ]
}
