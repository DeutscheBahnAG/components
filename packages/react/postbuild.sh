#! /bin/bash

cp -p package.json dist &&
for f in `ls *.md`; do cp -p $f dist; done &&
for f in `ls src/*.d.ts`; do cp -p $f dist; done &&
for f in `ls src/*/README.md`; do cp -p $f ${f/src/dist}; done &&
perl -i -pe 's#dist/index#index#' dist/package.json
perl -i -pe 's#dist/module/index#module/index#' dist/package.json
