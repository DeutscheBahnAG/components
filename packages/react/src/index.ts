export * as TransportchipIcons from './transportchip/icons';

export { default as Anchor } from './anchor';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as Container } from './container';
export { default as Copy } from './copy';
export { default as Layout } from './layout';
export { default as Link } from './link';
export { default as Loadingindicator } from './loadingindicator';
export { default as Logo } from './logo';
export { default as Modal } from './modal';
export { default as Notification } from './notification';
export { default as Passwordfield } from './passwordfield';
export { default as Pulse } from './pulse';
export { default as Select } from './select';
export { default as Status } from './status';
export { default as Textfield } from './textfield';
export { default as Theme } from './theme';
export { default as Time } from './time';
export { default as Title } from './title';
export { default as Track } from './track';
export { default as Transportchip } from './transportchip';
export { default as Triptime } from './triptime';
export { default as Triptimespan } from './triptimespan';
export { default as Radiobutton } from './radiobutton';
export { default as Accordion } from './accordion';
export { default as Accordionitem } from './accordionitem';
export { default as List } from './list';
