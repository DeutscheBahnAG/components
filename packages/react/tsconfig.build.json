{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "esModuleInterop": true,
    "jsx": "react",
    "lib": ["dom", "ES2020"],
    "noImplicitAny": false,
    "outDir": "dist",
    "target": "ES2015",
    "sourceMap": false
  },
  "include": ["src"],
  "exclude": ["**/*.test.*", "**/*-doc.*", "**/*.snap"]
}
