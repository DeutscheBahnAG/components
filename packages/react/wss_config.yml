#DISCLAIMER
#All sensible data is not persisted and deleted shortly after usage. So be assured we cope according to any law with your data!

# NOTE: see here how these configs are used https://git.tech.rz.db.de/ass/dependencychecker/whitesource/whitesource-dockered-scanner/-/blob/master/Docker/base/wrapper/scan

##################
#   WHITESOURCE  #
##################
wss_project_name="db-design-react"
wss_project_version="$CI_COMMIT_BRANCH"
  #wss_product_version="latest" DO NOT SET THIS IF USING wss_project_version as this overrides it

  # The following variables must be set as env variables in the Gitlab bahn-x group, they are valid for all subgroups
  #
  # wss_apiKey  -> your organization key, from Whitesource WebGUI:  Integrate > Organization > API key
  # wss_userKey -> CI tech-user key, as received initially via email, also found in 1pass

  # The following variables must be set as env variables in the Gitlab bahn-x subgroups like TMP, Analytics etc
  #
  # wss_product_name  -> set set this to one of our whitesource "products"  https://whitesource.int-app.comp.db.de/Wss/WSS.html#!allProducts
  # wss_product_token -> the products key, from Whitesource WebGUI:  Integrate > Product Tokens > Token

  # Npm specific config: If you have scoped dependencies, please add them according to the conventions below:
  # npmScopedDeps="@yourScope:registry=yourRegistryLink;@yourScope2:registry=yourRegistryLink2"
npmScopedDeps="@bahn-x:registry=https://npm.pkg.github.com"

  # Whitesource will scan your entire project by deafult.
  # Change this, if you want scan only a specific path.
wss_scan_folder=packages/react/

# You feel brave today?
# Great, use your own wss-unified-agent.config file as of here: https://whitesource.atlassian.net/wiki/spaces/WD/pages/489160834/Unified+Agent+Configuration+File+Parameters
# Tell us, where we will find it in your repo structure, starting from the root directory.
# no leading slash!!
#
# !! Please note that in this case all information (including project / product name and version) must be specified in this team-specific Whitesource config file and will no longer be read from wss_config.yml
# (apart from wss_scan_folder and wss_javaParams which is meta-information (i.e. not Whitesource specific) and therefore still needs to be specified).
#wss_specific_team_config=whitesource/wss-unified-agent.config
