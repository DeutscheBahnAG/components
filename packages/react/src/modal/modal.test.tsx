/* global document */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../button';
import Modal, { ModalSizes } from './modal';

const APP_ID = 'root-test-id';
const PORTAL_ID = 'portal-test-id';

const ModalWrapper: React.FC = ({ children }) => {
  const rootElement = document.getElementById(APP_ID);
  const portalElement = document.getElementById(PORTAL_ID);
  // @NOTE Elements need to be in DOM before render. There's probably a better way to do this.
  if (!rootElement) {
    const appElement = document.createElement('div');
    appElement.setAttribute('id', APP_ID);
    document.body.append(appElement);
  }
  if (!portalElement) {
    const element = document.createElement('div');
    element.setAttribute('id', PORTAL_ID);
    document.body.append(element);
  }
  return <div>{children}</div>;
};

ModalWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

describe('Modal Component', () => {
  beforeAll(() => {
    // @NOTE Elements need to be in DOM before render. There's probably a better way to do this.
    const appElement = document.createElement('div');
    appElement.setAttribute('id', APP_ID);
    document.body.append(appElement);

    const element = document.createElement('div');
    element.setAttribute('id', PORTAL_ID);
    document.body.append(element);
  });

  test('render required properties only', () => {
    const { baseElement } = render(
      <Modal isOpen appId={APP_ID} portalId={PORTAL_ID}>
        This is a basic render of required properties.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('render optional properties', () => {
    const { baseElement } = render(
      <Modal
        isOpen
        appId={APP_ID}
        portalId={PORTAL_ID}
        ariaDescribedBy="modal-desc"
        ariaLabelledBy="modal-label"
        className="some-class"
        closeText="Custom text"
        overlayClassName="some-overlay-class"
        title="Modal TITLE"
      >
        This is a basic render including optional properties.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('alertdialog role', () => {
    const { baseElement } = render(
      <Modal isOpen appId={APP_ID} portalId={PORTAL_ID} kind="alert">
        Modal should have alert role.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test.each(ModalSizes)('size %s', (size) => {
    const { baseElement } = render(
      <Modal isOpen appId={APP_ID} portalId={PORTAL_ID} size={size}>
        Modal should have size={size}.
      </Modal>,
      {
        wrapper: ModalWrapper,
      }
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('no close button', () => {
    const { baseElement } = render(
      <Modal isOpen appId={APP_ID} portalId={PORTAL_ID} enableCloseButton={false}>
        Modal without a close button.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('render PrimaryButton', () => {
    const { baseElement } = render(
      <Modal
        isOpen
        appId={APP_ID}
        portalId={PORTAL_ID}
        primaryButton={<Button variant="primary">Primary Button Here</Button>}
      >
        Modal with PrimaryButton.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('render SecondaryButton', () => {
    const { baseElement } = render(
      <Modal
        isOpen
        appId={APP_ID}
        portalId={PORTAL_ID}
        secondaryButton={<Button>Secondary Button Here</Button>}
      >
        Modal with SecondaryButton.
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
