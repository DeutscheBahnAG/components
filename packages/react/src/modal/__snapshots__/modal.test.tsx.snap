// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component alertdialog role 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="assertive"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="alertdialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal should have alert role.
          </main>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component no close button 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          />
          <main
            class="db-modal__content"
          >
            Modal without a close button.
          </main>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component render PrimaryButton 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal with PrimaryButton.
          </main>
          <div
            class="db-modal__actions"
          >
            <div
              class="db-modal__action db-modal__action--primary"
            >
              <button
                class="db-button db-button--primary db-button--default db-button--size-l"
                type="button"
              >
                Primary Button Here
              </button>
              <span
                class="db-inline-spacer"
              >
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component render SecondaryButton 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal with SecondaryButton.
          </main>
          <div
            class="db-modal__actions"
          >
            <div
              class="db-modal__action db-modal__action--secondary"
            >
              <button
                class="db-button db-button--primary db-button--default db-button--size-l"
                type="button"
              >
                Secondary Button Here
              </button>
              <span
                class="db-inline-spacer"
              >
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component render optional properties 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal some-overlay-class db-modal--open"
      >
        <div
          aria-describedby="modal-desc"
          aria-labelledby="modal-label"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m some-class db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Custom text"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Custom text
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
            <h2
              class="db-modal__title"
              id="ariaLabelledBy"
            >
              <span
                class="db-headline db-fs7 t:db-fs8 d:db-fs8"
              >
                Modal TITLE
              </span>
            </h2>
          </header>
          <main
            class="db-modal__content"
          >
            This is a basic render including optional properties.
          </main>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component render required properties only 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            This is a basic render of required properties.
          </main>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component size auto 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-auto db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal should have size=
            auto
            .
          </main>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component size l 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-l db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal should have size=
            l
            .
          </main>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component size m 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-m db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal should have size=
            m
            .
          </main>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;

exports[`Modal Component size s 1`] = `
<body
  class="db-body--db-modal-open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    id="root-test-id"
  />
  <div
    aria-live="polite"
    aria-relevant="additions removals"
    id="portal-test-id"
  >
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    />
    <div
      class="db-modal-portal"
    >
      <div
        class="db-modal db-modal--open"
      >
        <div
          aria-labelledby="modal-title"
          aria-modal="true"
          class="db-modal__dialog db-modal__dialog--width-s db-modal__dialog--open"
          role="dialog"
          tabindex="-1"
        >
          <header
            class="db-modal__header db-modal__header--collapse"
          >
            <button
              class="db-button db-button--hover-only db-button--square db-button--size-m db-modal__closebutton"
              title="Close the dialogue"
              type="button"
            >
              <svg
                class="db-icon"
                height="24"
                role="presentation"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 18c0-.281.094-.5.281-.688L10.594 12 5.28 6.687A1.03 1.03 0 015.031 6c0-.563.438-.969.969-.969.281 0 .5.094.688.25L12 10.594l5.313-5.313A.92.92 0 0118 5c.563 0 1 .438 1 1 0 .281-.094.5-.313.688L13.407 12l5.28 5.313A.844.844 0 0119 18a.98.98 0 01-.969 1 .915.915 0 01-.718-.313L12 13.407l-5.313 5.28A.915.915 0 015.97 19 .98.98 0 015 18z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                aria-hidden="false"
              >
                Close the dialogue
              </span>
            </button>
            <span
              class="db-inline-spacer"
            >
               
            </span>
          </header>
          <main
            class="db-modal__content db-modal__content--right-gap"
          >
            Modal should have size=
            s
            .
          </main>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
</body>
`;
